src/scheduler/resv_info.c

```diff
@@ -534,10 +534,10 @@ query_reservations(server_info *sinfo, struct batch_status *resvs)
 			}
 		}
 		resresv_arr[idx] = resresv;
+		resresv_arr[idx + 1] = NULL;
```

Comment:
It's clear what is going on.  I just thought it was a little weird.  I'm fine with it.  Rebase and squash and I'll merge your code.